Modules/mathmodule.c

```diff
@@ -2231,14 +2226,12 @@ loghelper(PyObject* arg, double (*func)(double))
         }
 
         x = PyLong_AsDouble(arg);
-        if (x == -1.0 && PyErr_Occurred()) {
-            if (!PyErr_ExceptionMatches(PyExc_OverflowError))
-                return NULL;
+        if (PyErr_Occurred()) {
```

Comment:
What other value can `PyLong_AsDouble()` return on error? It should be fixed to always return -1.0 on error.

Reply:
No, the C-API is correct here :)  The point of this change (and similar below on L2241) is that the case "there is an error AND x is -1.0" is inaccessible: we already accept only *positive* ints (``_PyLong_IsPositive`` test above).

Reply:
Can `PyLong_AsDouble()` fail? If not then this branch can be removed.

The check `x == -1.0` is not for successfully returned -1.0 value, it is an optimization for error check, because `x == -1.0` is more efficient than `PyErr_Occurred()`. If `arg` is always positive, you can remove `PyErr_Occurred()` here, but I would not do this without consultation with @mdickinson.

Reply:
> Can PyLong_AsDouble() fail?

Sure.

> The check x == -1.0 is not for successfully returned -1.0 value, it is an optimization for error check

Yep, that does make sense, thanks.  ``PyErr_Occurred()`` should be removed instead.

I think we could be safe here: it's documented that ``PyLong_AsDouble()`` does return ``-1.0`` on error.  Same seems to be true for the ``_PyLong_Frexp()``, from reading it's code (but, perhaps, we could add an assert here).